pkg/cloud/awsprovider.go

```diff
@@ -57,22 +57,31 @@ func (aws *AWS) PricingSourceStatus() map[string]*PricingSource {
 
 	sps := &PricingSource{
 		Name: SpotPricingSource,
+		Enabled: true,
 	}
-	sps.Error = ""
```

Comment:
View this diff side-to-side; what happens here is I'm wrapping the existing logic into the `else` block of the `if !aws.SpotRefreshEnabled` statement. This makes it so if the flag is false we simply set `available:false, enabled:false` and a hardcoded message for the `error` field.